### ISSUE TITLE
Vault api sending multiple errors as a single string in json

### ISSUE THREAD

**Issue by pjain-fastly** [hashicorp/vault#28792]
**Describe the bug**
Starting with version 1.17.*, Vault API is sending multiple errors in a single json. This new json format is hard to parse because the errors are lumped together in a single string, separated by newline characters, rather than as discrete items in an array. This makes string parsing complex since we cannot simply iterate over the items.

**To Reproduce**
1. Write a secret
2. Try reading the same secret with a token which doesn't have permissions to read it.
3. See the error:

API: `{"errors":["2 errors occurred:\n\t* permission denied\n\t* invalid token\n\n"]}`

CLI: 
`
* 2 errors occurred:
	* permission denied
	* invalid token
`

**Expected behavior**
Ideally, we would like to see the error messages be sent in the standard json format as separate items like below:
`"errors": ["permission denied", "invalid token"]`

**Environment:**
* Vault Server Version: We tested this on version 1.17.0 and 1.17.6
* Vault CLI Version (retrieve with `vault version`): Vault v1.16.1
* Server Operating System/Architecture: Kubernetes pods

Vault server configuration file(s):
We don't think this issue has nything to do with config file. If needed we can share it at a later stage based on your troubleshooting.

**Additional context**
This issue breaks our automation and how we handle authentication errors.

**Comment by joeshaw** [comment]
Versions 1.15.5 and earlier (not sure about 1.16.x) returned just "permission denied" in this case.

**Comment by mpalmi** [comment]
I was able to bisect this to commit 74abae6f017cd7262aaf239355569d70e6e2cdc7. It seems reasonable to provide these fields as separate strings in the multierror response, but I'll need to check how that's handled elsewhere.

More to come...

**Comment by mpalmi** [comment]
Okay, following the trail a bit further, it looks like there is a pre-existing report of this issue as it relates to the PR which introduced the breaking change over in #27772.

@pjain-fastly @joeshaw any objections to me closing this issue out as a duplicate?
